jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/component/SearchField.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2022 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.searchflowui.component;
+
+import com.vaadin.flow.component.Key;
+import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import com.vaadin.flow.router.QueryParameters;
+import io.jmix.core.Messages;
+import io.jmix.flowui.DialogWindows;
+import io.jmix.flowui.Notifications;
+import io.jmix.flowui.ViewNavigators;
+import io.jmix.flowui.component.UiComponentUtils;
+import io.jmix.flowui.component.textfield.TypedTextField;
+import io.jmix.flowui.view.DialogWindow;
+import io.jmix.flowui.view.OpenMode;
+import io.jmix.search.SearchProperties;
+import io.jmix.search.searching.SearchStrategy;
+import io.jmix.searchflowui.view.result.SearchResultsView;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.jmix.searchflowui.view.result.SearchResultsView.*;
+
+//todo add studio component annotation
+public class SearchField extends TypedTextField<String> {
```

Comment:
I doubt you need all the APIs from TextField. It seems that `CustomField` is more suitable base class